client/src/main/java/io/pravega/client/tables/impl/KeyValueTableFactoryImpl.java

```diff
@@ -38,7 +39,7 @@
             @NonNull String keyValueTableName, @NonNull Serializer<KeyT> keySerializer,
             @NonNull Serializer<ValueT> valueSerializer, @NonNull KeyValueTableClientConfiguration clientConfiguration) {
         val kvt = new KeyValueTableInfo(this.scope, keyValueTableName);
-        val provider = DelegationTokenProviderFactory.create(this.controller, kvt.getScope(), kvt.getKeyValueTableName());
+        val provider = DelegationTokenProviderFactory.create(this.controller, kvt.getScope(), kvt.getKeyValueTableName(), AccessOperation.WRITE);
```

Comment:
This isn't so simple. KeyValueTable clients can be used for both reading or writing. What if someone just wants to read from KVTs?

Reply:
@andreipaduroiu how do you suggest we handle this? 

One possibility I see is to let the caller specify the intended access operation (`READ`, `WRITE`, or `READ_WRITE`). If left unspecified, default it to `READ_WRITE` (both read and `WRITE`)? 

Alternatively, we could leave it as-is and enhance it later when we support distinguishing write-only from read_write. As of now, both `WRITE` and `READ_WRITE`  are interpreted as `READ_UPDATE` (both reads and writes) on the server-side. 
https://github.com/pravega/pravega/blob/6d27990a04eea7ebdc1f132ae3a7942d3beb83ed/shared/security/src/main/java/io/pravega/shared/security/auth/PermissionsHelper.java#L30-L31

Reply:
one alternative is -- client requests a read `and` read_update token lazily. 
for example, when any "create/update/remove" api on kvt is called, if a write token exists, its used, else its explicitly requested.
similarly when a read/iterate api is called on kvt, a read token is requested once and used henceforth. 

if users are unauthorized for updates, their attempts to update the table may fail but they will still be able to read.

Reply:
I'm not sure what the use case would be, but a generic one would be that one an application A maintains a table that another application B consumes data from. A read-only table would allow a client only to perform gets on an existing table, while read_update would enable it to perform all operations.